### ISSUE TITLE
Dashboard with ranking (gamification) and learning graph on the profile page

### ISSUE THREAD

**Issue by adachiu** [codebuddies/codebuddies#61]
Features to be considered in the future.

**Ranking System**
To incentivize people to show up to scheduled hangouts, we could implement a ranking point system.
Showing up at a scheduled hangout = +1 point
Signed up, but didn't show up = -1 point
(Or we could increase the penalty for "no shows" to -5 points or more...)

The ranking levels could be based on hangouts completed in increments of 25.
Ex. 25 Hangouts completed = Rank level 1
(ignore the wrong rank level in the mock 😝  ..that would suck to still be rank level 1 after completed 218 hangouts LOL)

**Learning Graph (similar to Github's graphical streak)**
To encourage people to log their learnings/accomplishments, we could have a graphical streak system.

The darker colored square = more than 1 learning/accomplishment logged that day. The #TodayILearned section shows only one date entry to keep the page clean. If the user has more than 1 learning/accomplishment in one day, it would show altogether as one date entry (as shown in this example mock). Clicking on the squares in the graph would refresh the #TodayILearned entry to show the specific date's learning/accomplishment. Some ideas on coding the graph: http://bl.ocks.org/tjdecke/5558084

![dash](https://cloud.githubusercontent.com/assets/13413986/13837166/ab92c52c-ebc5-11e5-92d8-ae56eb01cba1.jpg)

**Comment by lpatmo** [comment]
Features we can remove from the MVP:
- "Rank level 1"
- Complete/Incomplete
- 2 day learning streak and associated grid

**Comment by lpatmo** [comment]
After #829 is fleshed out more, let's come back to the profile page and add at least:

- a "total points accumulated" number for the user at the top of their profile page
- consider implementing the learning streak graph

**Comment by billglover** [comment]
Closing this issue as stale. This doesn't mean we have ruled out gamification. If this feature is still of interest, please open a new issue and include a reference to this issue by including issue number #61 in the description.
